src/poly.rs

```diff
@@ -200,3 +200,32 @@ impl<'a, F: Field, B: Basis> Mul<F> for Polynomial<F, B> {
         self
     }
 }
+
+/// Describes a relative location in the evaluation domain; applying a rotation
+/// by i will rotate the vector in the evaluation domain by i.
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, Ord, PartialOrd)]
```

Comment:
It's best not to derive these things unless and until we need them so that we can later expand the API contract; there may be reasons (that we cannot guess now) that would make it problematic for a user to be able to sort and hash these rotations. Anecdote is that in bellman being able to `Hash` a `Variable` would have allowed a user to create non-deterministic hashmaps in circuit synthesis code, which meant that circuit synthesis would be non-deterministic.

Reply:
I would support `Copy, Clone, Debug` though.

Reply:
Also, I'm not entirely sure it makes sense to sort rotations because Rotation(a) = Rotation(b) could be true for a particular vector due to length overflow.